component/src/main/java/org/exoplatform/rhmanagement/integration.notification/MailTemplateProvider.java

```diff
@@ -0,0 +1,257 @@
+      NotificationInfo first = notifications.get(0);
+
+      String language = getLanguage(first);
+      TemplateContext templateContext = new TemplateContext(first.getKey().getId(), language);
+      //
+      Identity receiver = CommonsUtils.getService(IdentityManager.class).getOrCreateIdentity(OrganizationIdentityProvider.NAME, first.getTo(), true);
+      templateContext.put("FIRST_NAME", encoder.encode(receiver.getProfile().getProperty(Profile.FIRST_NAME).toString()));
+      templateContext.put("FOOTER_LINK", LinkProviderUtils.getRedirectUrl("notification_settings", receiver.getRemoteId()));
+      
+      try {
+        writer.append(buildDigestMsg(notifications, templateContext));
+      } catch (IOException e) {
+        ctx.setException(e);
+        return false;
+      }
+      return true;
+    }
+
+    protected String buildDigestMsg(List<NotificationInfo> notifications, TemplateContext templateContext) {
+      EntityEncoder encoder = HTMLEntityEncoder.getInstance();
+      
+      Map<String, List<NotificationInfo>> map = new HashMap<String, List<NotificationInfo>>();
+      for (NotificationInfo notif : notifications) {
+        String activityID = notif.getValueOwnerParameter(NotificationUtils.ACTIVITY_ID);
+        List<NotificationInfo> tmp = map.get(activityID);
+        if (tmp == null) {
+          tmp = new LinkedList<NotificationInfo>();
+          map.put(activityID, tmp);
+        }
+        tmp.add(notif);
+      }
+      
+      StringBuilder sb = new StringBuilder();
+      for (String activityID : map.keySet()) {
+        List<NotificationInfo> notifs = map.get(activityID);
+        NotificationInfo first = notifs.get(0);
+        String vacationUrl = first.getValueOwnerParameter(NotificationUtils.VACATION_URL);
+        if (notifs.size() == 1) {
+          templateContext.digestType(ElementType.DIGEST_ONE.getValue());
+        } else {
+          templateContext.digestType(ElementType.DIGEST_MORE.getValue());
+        }
+        sb.append("<li style=\"margin:0 0 13px 14px;font-size:13px;line-height:18px;font-family:HelveticaNeue,Helvetica,Arial,sans-serif\"><div style=\"color: #333;\">");
+        String digester = TemplateUtils.processDigest(templateContext);
+        sb.append(digester);
+        sb.append("</div></li>");
+      }
+      return sb.toString();
+    }
+  }
+
+
+  public static String getExcerpt(String str, int len) {
+    if (str == null) {
+      return "";
+    } else if (str.length() > len) {
+      str = str.substring(0, len);
+      int lastSpace = str.lastIndexOf(" ");
+      return ((lastSpace > 0) ? str.substring(0, lastSpace) : str) + "...";
+    } else {
+      return str;      
+    }
+  }
+
+
+  protected String getDate(String date, String userName) {
+    if (date != null) {
+      Date date_ = new Date(Long.parseLong(date));
+      return org.exoplatform.rhmanagement.integration.notification.TemplateUtils.format(date_,getUserTimezone(userName));
+    } else {
```

Comment:
org.exoplatform.rhmanagement.integration.notification.TemplateUtils should extends org.exoplatform.commons.notification.template.TemplateUtils and be used instead of it.